src/cf.data.pre

```diff
@@ -4979,10 +4979,32 @@ DOC_START
 		[http::]<pt	Peer response time in milliseconds. The timer starts
 				when the last request byte is sent to the next hop
 				and stops when the last response byte is received.
-		[http::]<tt	Total time in milliseconds. The timer
-				starts with the first connect request (or write I/O)
-				sent to the first selected peer. The timer stops
-				with the last I/O with the last peer.
+		[http::]<tt	Total peering time in milliseconds.
+
+				The timer starts when Squid decides to forward the request (to
+				an origin server or cache_peer) and peer selection begins. The
+				timer stops when relevant forwarding activities (including any
+				retries) end.
```

Comment:
AFAICT, current PR code does not quite match this documentation because, besides the "natural" end of forwarding activities resulting in a stopPeerClock() call (as documented), totalResponseTime() may also call stopPeerClock() (resulting in undocumented outcomes). Moreover, after totalResponseTime() calls stopPeerClock(), subsequent stopPeerClock(false) calls have no effect, resulting in smaller-than-actual `%<tt` values logged!

N.B. Supplying `%<tt` to an external ACL helper may trigger totalResponseTime() calls before forwarding ends AFAICT.

Reply:
> Request::hier is copied into ALE::hier

This 'copying' approach, as we discussed, may be dangerous if there are several HttpRequest objects created before one of them is finally copied. However, I could not find whether this scenario is possible. I ran several tests producing revalidation (sub) transactions, including both 304 and 200 responses - in all these tests exactly one HttpRequest object  was created.

Reply:
The only problem with "copying" approach that we know of right now is that `%<tt` does not work outside of access.log context (i.e. after prepareLogWithRequestDetails()) has been called, right? For example, it does not work with external ACLs, adaptation_meta directive, and (if they support %cods) annotate_transaction ACLs.

Reply:
Right, I described this problem in the PR description.

Reply:
Thank you. To fix this bug, we should move HttpRequest::hier (or its portions, including peeringTime_) to ALE. You have already commented on that [earlier](https://github.com/measurement-factory/squid/pull/249/files#r1546418065), but now we know of at least one specific bug created by the current two-hier design.

I agree with your earlier opinion -- we should not do that move in _this_ PR (unless we are forced to). 

If there is no simple fix, we will leave this particular bug as is for now, documented in the PR description only.

Is it possible to (at least partially) fix this bug by adjusting formatting code to look at the HttpRequest (in ALE; instead of ALE::hier) if (and only if) ALE::hier::peeringTime_ has not been set? If that small change can fix `%<tt` in all or most cases, then please do that in this PR.

Reply:
> Is it possible to (at least partially) fix 

Done (94bb99b9).